src/utils/mod.rs

```diff
@@ -10,6 +10,11 @@ use crate::errors::RvError;
 
 pub mod cert;
 pub mod key;
+pub mod salt;
+pub mod cidr;
+pub mod sockaddr;
+pub mod ipaddr;
+pub mod unixsock;
```

Comment:
The names of these two modules are very confusing. 'Unix socket (`mod unixsock`)' is not in the same semantic context with the term 'IP address (`mod ipaddr`)'. You should better call them `unixsockaddr` and `ipsockaddr`, and rename the relevant filenames.

Then this can abstract the hierarchy that two types of socket address have common methods like `contains()` and `equal()` which are defined in trait `SockAddr`.

Reply:
> The names of these two modules are very confusing. 'Unix socket (`mod unixsock`)' is not in the same semantic context with the term 'IP address (`mod ipaddr`)'. You should better call them `unixsockaddr` and `ipsockaddr`, and rename the relevant filenames.
> 
> Then this can abstract the hierarchy that two types of socket address have common methods like `contains()` and `equal()` which are defined in trait `SockAddr`.

The implementation of ipaddr and unixsock is based on the one found at https://github.com/hashicorp/go-sockaddr, and there should be no issues with semantics.
<img width="992" alt="image" src="https://github.com/Tongsuo-Project/RustyVault/assets/7484739/94f3a0d0-fe1a-406c-b3c6-ed57c247dfd8">

Reply:
fixed